### PR TITLE
Refactor tests to use phpunit properly with standardized bootstrappin…

### PR DESCRIPTION
**Pull request by ircmaxell**
…g and managing phpunit via composer

This standardizes PHPUnit, and removes the need for GPG and downloads to actually test the suite.

Plus it allows for simple tests to be run with `$ vendor/bin/phpunit` by itself. This allows for changes to be tested far easier than requiring all of the permutations every time.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -23,4 +23,4 @@ install:
 - composer update
 - chmod +x ./tests/phpunit.sh
```
Comment (jdreesen) [inline .travis.yml]: This should be `- chmod +x ./phpunit.sh`, I think.

Comment (ircmaxell) [inline .travis.yml]: Yup, good catch

File: composer.json [hunk 2]
```diff
@@ -23,6 +23,9 @@
     "require": {
         "php": ">=5.2.0"
     },
+    "require-dev": {
+        "phpunit/phpunit": "4.*"
```
Comment (GrahamCampbell) [inline composer.json]: Maybe consider `4.*|5.*` because, officially, phpunit 4 is marked as not compatible with php 7, only phpunit 5 is.

Comment (ircmaxell) [inline composer.json]: yeah, the only reason I avoided 5 is because this lib is designed to run on <= 5.6, including 5.5, 5.4 and 5.3...

Comment (GrahamCampbell) [inline composer.json]: php 7 is tested on travis

Comment (GrahamCampbell) [inline composer.json]: i assume that's to assert that the behaviour is the same as the actual php implementation

**Pull request merged**